source/enemies/earthlyknight.js

```diff
@@ -0,0 +1,80 @@
+const { EnemyTemplate } = require("../classes/index.js");
+const { dealDamage, removeModifier, changeStagger } = require("../util/combatantUtil.js");
+const { isBuff } = require("../modifiers/_modifierDictionary");
+const { selectRandomFoe, selectNone, selectAllFoes } = require("../shared/actionComponents.js");
+const { spawnEnemy } = require("../util/roomUtil.js");
+const { getEmoji } = require("../util/elementUtil.js");
+
+const asteroid = require("./asteroid.js")
+
+const PATTERN = {
+  "Call Asteroid": "Tremor Smash",
+  "Damping Wallop": "random",
+  "Tremor Smash": "random"
+}
+function earthlyKnightPattern(actionName) {
+  return PATTERN[actionName]
+}
+
+module.exports = new EnemyTemplate("Earthly Knight",
+  "Earth",
+  250,
+  100,
+  "6",
+  0,
+  "Tremor Smash",
+  false
+).addAction({
+  name: "Damping Wallop",
+  element: "Earth",
+  description: `Inflict ${getEmoji("Earth")} damage and remove a buff`,
+  priority: 0,
+  effect: (targets, user, isCrit, adventure) => {
+    let damage = user.getPower() + 75;
+    changeStagger(targets, "elementMatchFoe");
+    let resultString = ""
+    for (target of targets) {
+      resultString += dealDamage([target], user, damage, false, user.element, adventure);
+      const targetBuffs = Object.keys(target.modifiers).filter(modifier => isBuff(modifier));
+      console.log(targetBuffs);
+      const buffIndex = adventure.generateRandomNumber(targetBuffs.length, "battle");
+      const rolledDebuff = targetBuffs[buffIndex];
+      const wasRemoved = removeModifier([target], { name: rolledDebuff, stacks: "all" }).length > 0;
+      if (wasRemoved) {
```

Comment:
This conditional will always evaluate to `true` because even if `removeModifier` returns an empty array, empty arrays are truthy in JS.